src/maingoblint.ml

```diff
@@ -508,6 +508,24 @@ let do_gobview () =
     if Sys.file_exists js_file then (
       let save_run = GobConfig.get_string "save_run" in
       let run_dir = if save_run <> "" then save_run else "run" in
+      (* copy relevant c files to gobview directory *)
+      let file_dir = Fpath.((Fpath.v run_dir) / "files") in
+      GobSys.mkdir_or_exists file_dir;
+      let file_loc = Hashtbl.create 113 in
+      let copy_rem p =
+        let _, name = BatFile.open_temporary_out ~prefix:(Fpath.filename p) ~suffix:".c" ~temp_dir:(Fpath.to_string file_dir) () in
+        let gobview_path = match Fpath.relativize ~root:(Fpath.v run_dir) (Fpath.v name) with
+          | Some p -> Fpath.to_string p
+          | None -> failwith "The gobview directory should be a prefix of the paths of c files copied to the gobview directory" in
+        Hashtbl.add file_loc (Fpath.to_string p) gobview_path;
+        BatFile.write_lines name (BatFile.lines_of (Fpath.to_string p)) in
+      let paths = Cil.foldGlobals cilfile (
+          fun acc g -> match g with GFun (_,loc) -> if not (List.mem loc.file acc) then loc.file :: acc else acc | _ -> acc) [] in
+      List.iter copy_rem (List.map Fpath.v paths);
```

Comment:
Maybe we can use `Fileutils.cp` here, instead of doing it manually linewise

Reply:
`Fileutils.cp`? I don't see us having anything like that.

Could just use `create_symlink` anyway, which this logic does for all the other Gobview files as well. Yes, that means changing the files in their original locations changes the file shown in Gobview for the old results, but that's already the case because everything else is symlinked as well. To be fully persistent, everything needs to be copied instead of symlinking.

Reply:
> `Fileutils.cp`? I don't see us having anything like that.

It is an external library that is part of #995

Reply:
I thought symlinking might not be so good, because one would like to forbid following any symlink and really limit the file access to the served directory to avoid symlink attacks.

Reply:
> Could just use create_symlink anyway, which this logic does for all the other Gobview files as well.

Most other files for Gobview (e.g. the cil file, configs, solver stats) are marshalled directly to the Gobview directory. So they are not symlinked, but there exists a separate copy of them just for Gobview. The only ones that I find that are symlinked are some javascript file for the monaco editor, the `index.html`, `main.js` and one `.ttf` file.